File: src/main/java/dev/slne/discord/listener/message/MessageCreatedListener.java
package dev.slne.discord.listener.message;

import dev.slne.discord.annotation.DiscordListener;
import dev.slne.discord.spring.service.ticket.TicketService;
import dev.slne.discord.ticket.Ticket;
import dev.slne.discord.ticket.message.TicketMessage;
import javax.annotation.Nonnull;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import org.springframework.beans.factory.annotation.Autowired;

/**
 * The type Message created listener.
 */
@DiscordListener
public class MessageCreatedListener extends AbstractMessageListener<MessageReceivedEvent> {

  @Autowired
  public MessageCreatedListener(TicketService ticketService) {
    super(ticketService);
  }

  @Override
  public void onMessageReceived(@Nonnull MessageReceivedEvent event) {
    processEvent(event);
  }

  @Override
  protected void handleEvent(MessageReceivedEvent event, Ticket ticket) {
    if (event.getMessage().isWebhookMessage()) {
      return;
    }

    ticket.addTicketMessage(TicketMessage.fromTicketAndMessage(ticket, event.getMessage()));
  }
}

File: src/main/kotlin/dev/slne/discord/listener/message/MessageCreatedListener.kt
package dev.slne.discord.listener.message

import dev.minn.jda.ktx.events.listener
import dev.slne.discord.extensions.ticketOrNull
import dev.slne.discord.persistence.service.ticket.TicketService
import dev.slne.discord.util.toTicketMessage
import jakarta.annotation.PostConstruct
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import org.springframework.stereotype.Component

@Component
class MessageCreatedListener(private val jda: JDA, private val ticketService: TicketService) {

    @PostConstruct
    fun registerListener() {
        jda.listener<MessageReceivedEvent> { event ->
            if (event.message.isWebhookMessage) {
                return@listener
            }

            val ticket = event.channel.ticketOrNull() ?: return@listener

            val message = event.message.toTicketMessage()
            ticket.addMessage(message)

            ticketService.saveTicket(ticket)
        }
    }
}
